File: app/src/main/java/com/sonu/app/splash/data/network/unsplashapi/ApiEndpoints.java
package com.sonu.app.splash.data.network.unsplashapi;

/**
 * Created by amanshuraikwar on 20/12/17.
 */

public interface ApiEndpoints {

    int PER_PAGE = 30;

    String BASE_ENDPOINT = "https://api.unsplash.com/";

    String GET_ALL_PHOTOS = BASE_ENDPOINT + "photos?page=%s&order_by=%s&per_page="+PER_PAGE;

    String GET_CURATED_PHOTOS = BASE_ENDPOINT + "photos/curated?page=%s&order_by=%s&per_page="+PER_PAGE;

    String GET_PHOTO_DESCRIPTION = BASE_ENDPOINT + "photos/%s";

    String SEARCH_PHOTOS = BASE_ENDPOINT + "search/photos?query=%s&page=%s&per_page="+PER_PAGE;

    String GET_USER_DESCRIPTION = BASE_ENDPOINT + "users/%s";

    String GET_USER_PHOTOS = BASE_ENDPOINT + "users/%s/photos?page=%s&order_by=%s&per_page="+PER_PAGE;

    String SEARCH_USERS = BASE_ENDPOINT + "search/users?query=%s&page=%s&per_page="+PER_PAGE;

    String GET_ALL_COLLECTIONS = BASE_ENDPOINT + "collections?page=%s&per_page="+PER_PAGE;

    String GET_FEATURED_COLLECTIONS = BASE_ENDPOINT + "collections/featured?page=%s&per_page="+PER_PAGE;

    String GET_CURATED_COLLECTIONS = BASE_ENDPOINT + "collections/curated?page=%s&per_page="+PER_PAGE;

    String SEARCH_COLLECTIONS = BASE_ENDPOINT + "search/collections?query=%s&page=%s&per_page="+PER_PAGE;

    String COLLECTION_DESCRIPTION = BASE_ENDPOINT + "collections/%s";

    String GET_COLLECTION_PHOTOS = BASE_ENDPOINT + "collections/%s/photos?page=%s&per_page="+PER_PAGE;
}

File: app/src/main/java/io/github/amanshuraikwar/splash/data/network/ApiEndpoints.kt
package io.github.amanshuraikwar.splash.data.network


/**
 * API endpoints for the remote unsplash.com api.
 *
 * @author Amanshu Raikwar
 * Created by amanshuraikwar on 30/04/18.
 */

@Suppress("MemberVisibilityCanBePrivate")
interface ApiEndpoints {

    companion object {

        const val PER_PAGE = 30
        const val STATS_QUANTITY = 30
        const val STATS_RESOLUTION = "days"

        const val BASE_ENDPOINT = "https://api.unsplash.com/"

        const val GET_ALL_PHOTOS = "photos?"

        const val GET_CURATED_PHOTOS = "photos/curated?"

        const val GET_PHOTO = "photos/{id}"

        const val SEARCH_PHOTOS = "search/photos?"

        const val GET_USER = "users/{id}"

        const val GET_USER_PHOTOS = "users/{id}/photos?"

        const val GET_USER_COLLECTIONS = "users/{id}/collections?"

        const val SEARCH_USERS = "search/users?"

        const val GET_ALL_COLLECTIONS = "collections?"

        const val GET_FEATURED_COLLECTIONS = "collections/featured?"

        const val GET_CURATED_COLLECTIONS = "collections/curated?"

        const val SEARCH_COLLECTIONS = "search/collections?"

        const val GET_COLLECTION= "collections/{id}"

        const val GET_COLLECTION_PHOTOS = "collections/{id}/photos?"

        const val GET_PHOTO_STATISTICS = "photos/{id}/statistics?"
    }
}